ast-visual-studio-extension/CxExtension/CxWindowControl.xaml.cs

```diff
@@ -138,6 +141,26 @@ private void OnChangeProject(object sender, SelectionChangedEventArgs e)
         {
             cxToolbar.ProjectsCombobox.OnChangeProject(sender, e);
         }
+        private async void OnProjectTextChanged(object sender, KeyEventArgs e)
+        {
+            typingCts?.Cancel();
+            typingCts?.Dispose();
+
+            typingCts = new CancellationTokenSource();
+            var token = typingCts.Token;
```

Comment:
why are we creating here canaclation token?

Reply:
The cancellation token is used here to prevent unnecessary operations when the user types quickly.